Loader component and special Apple fixes

DimaMklk opened the pull request.

Commit:
Loader component and new Main Star at Landing Page

Commit:
Loader filter's moz fix

Commit:
Main Star timing function fix

barbarosh commented on client/modules/App/App.js:
```diff
@@ -1,16 +1,20 @@
 // Import Actions
 import { authenticated } from './AppActions';
-import callApi from '../../util/apiCaller';
+import callApi, { isLoggedIn } from '../../util/apiCaller';
```
Где ті используешь isLoggedIn ?

DimaMklk replied:
@barbarosh я удалил с пятой строки и объеденил два импорта в одни. я не импортировал эту функцию, и не проверял, используеться ли она

barbarosh replied:
Ага нашел )

barbarosh commented on client/modules/App/pages/LandingPage/LandingPage.js:
```diff
@@ -70,15 +71,19 @@ export class LandingPage extends Component {
           <div className={styles.columnLeft}>
             <div className={styles.radarChart}>
+              <div className={styles.mainStar}>
+                <Loader />
```
Не пойму как это обудет віглядить

barbarosh commented on client/modules/App/pages/LandingPage/LandingPage.scss:
```diff
@@ -19,6 +41,26 @@
+          width: 100%;
+          position: absolute;
+
+          .userPhoto {
```
Если используем бем то давай уже использовать названия по спецификации.

Commit:
ClassNames BEM adjusting

Commit:
Loader positioning and Apple Devices fixes

! need npm install
Apple fixes for TalentIcon, LandingPage RadarCharts main star and DiscSlider

barbarosh commented on client/components/TalentIcon/TalentIcon.js:
```diff
@@ -47,13 +48,21 @@ export default class TalentIcon extends Component {
+            {
+              isAppleDevice()
+              ?
+              <div className={styles['apple-liquid-container']}>
```
Вынеси в отдельные функции

barbarosh commented on client/modules/App/App.js:
```diff
@@ -30,7 +34,15 @@ export class App extends Component {
 
   render() {
-    return this.props.children;
+    if (this.state.isMounted && typeof window !== 'undefined' && window.d3) {
```
Этой проверки быть не должно. Если что-то случится с д3.
то приложение вообще не покажется.

barbarosh commented on client/modules/User/components/DiscItem/DiscItem.js:
```diff
@@ -1,38 +1,59 @@
+            [styles['apple-controls-middle']]: this.props.value === undefined,
+            [styles['apple-controls-higher']]: this.props.value === 1
+          })}>
+            <Tooltip disabled={this.props.unavailableValue !== 0} tip="Select at least one that you are higher at">
```
В твоем случае нужно вынести фразы для тип в отдельную переменную.

barbarosh commented on client/util/generalHelpers.js:
```diff
@@ -1,5 +1,15 @@
 }
+
+export function isAppleDevice() {
+  if (browser.name === 'ios' || browser.name === 'safari' || browser.name === undefined) {
```
Изменить тело функции на return ((browser.name === 'ios') || (browser.name === 'safari') || browser.name === undefined));

Commit:
Review Fixes

- TalentIcon renderLiquidContainer function
- Removing d3 loading check at App.js
- Tip variables for DiscItem
- Refactoring isAppleDevice helper function

barbarosh commented on client/modules/User/components/DiscItem/DiscItem.js:
```diff
@@ -1,38 +1,61 @@
-          <div className={styles.sliderContainer}>
-            <RangeSlider value={this.props.value}/>
+        {
+          isAppleDevice()
```
Тоже самое раздели на две функции.

Commit:
TalentIcon and DiscItem subrender functions refactoring
